### PR TITLE
Trace code execution across actix queues for PeerManagerActor->PeerActor messages

### PR DESCRIPTION
**Pull request by nikurt**
When an actix actor handles a message from its queue, it does that outside of any spans.
This PR attaches the context of the requester to the request. Thus making the request handling seem like a direct child of spans of the requester.
In the future we'd like to trace execution across process boundaries. For example, to see a partial encoded chunk request made by process A, its handling by process B, and receiving a response by process A again. However, to make this possible we first need to trace execution within the process across the actix system.

### CODE REVIEW

File: core/o11y/src/lib.rs [hunk 1]
```diff
@@ -58,13 +58,26 @@ pub struct DefaultSubscriberGuard<S> {
     writer_guard: tracing_appender::non_blocking::WorkerGuard,
 }
 
+#[derive(Copy, Clone, Debug, clap::ArgEnum)]
+pub enum OpenTelemetryLevel {
```
Comment (nagisa) [inline core/o11y/src/lib.rs:62]: At this point I'd just allow specifying `LevelFilter` here. There's no real motivation that I can see to prohibit people from filtering to e.g. `INFO` level if they wished to do so for whatever reason.

**Review by nagisa — APPROVED** [review]
Overall nothing stands out as particularly wrong, so I'm happy to see this implemented and merged. However, I think it is worth replicating here what I said on Zulip about `set_parent` vs `add_link` – the former implies a hierarchical relationship whereas the latter – a causal one. I'm not super familiar with how our RPC is set up, but the `parent` vs `link` should be used based on that concept.

You mentioned that Jaeger doesn't really display `add_link` links in any meaningful way. I think it is worthwhile to investigate why that's the case, especially if we do end up instrumenting causal relationships (as would be necessary for e.g. cross-contract calls.)

File: core/o11y/src/lib.rs (hunk 1 above)
Comment (nikurt) [reply core/o11y/src/lib.rs:63]: Added an `INFO` value to `OpenTelementryLevel`.
Not sure what you mean. `LevelFilter` is a struct and can't be directly turned it into a command line flag.
Fun fact, `tracing::Level` is a struct, but `log::Level` is an enum.

**Comment by nikurt** [comment]
> `add_link` vs `set_parent`

Let's go with `set_parent` because it has a much better UI support. Added a TODO to revisit this choice eventually.

File: chain/network/src/peer_manager/peer_manager_actor.rs [hunk 2]
```diff
@@ -528,21 +534,32 @@ impl PeerManagerActor {
...
-                                network_metrics,
-                                &act.connected_peers,
-                                SendMessage {
-                                    message: PeerMessage::SyncRoutingTable(sync_routing_table),
-                                },
-                            )
+                .map(move |response, act, _ctx| {
+                    let _span = tracing::trace_span!(
+                        target: "network",
+                        parent: &span,
```
Comment (matklad) [inline chain/network/src/peer_manager/peer_manager_actor.rs:540]: This one I believe is buggy, see https://near.zulipchat.com/#narrow/stream/328640-temp-stakewars-prep/topic/shardnet.20debug for details.

**Pull request merged**